h5io/tests/test_io.py

```diff
@@ -151,24 +150,74 @@ def test_object_diff():
     pytest.raises(RuntimeError, object_diff, object, object)
 
 
-def test_numpy_values():
-    tempdir = _TempDir()
-    test_file = op.join(tempdir, 'test.hdf5')
+def test_numpy_values(tmpdir):
+    """Test NumPy values."""
+    test_file = op.join(str(tmpdir), 'test.hdf5')
     for cast in [np.int8, np.int16, np.int32, np.int64, np.bool_,
                  np.float16, np.float32, np.float64]:
         value = cast(1)
         write_hdf5(test_file, value, title='first', overwrite='update')
         assert_equal(read_hdf5(test_file, 'first'), value)
 
 
-def test_multi_dim_array():
+def test_multi_dim_array(tmpdir):
+    """Test multidimensional arrays."""
     rng = np.random.RandomState(0)
     traj = np.array([rng.randn(2, 1), rng.randn(3, 1)])
-    tempdir = _TempDir()
-    test_file = op.join(tempdir, 'test.hdf5')
+    test_file = op.join(str(tmpdir), 'test.hdf5')
     write_hdf5(test_file, traj, title='first', overwrite='update')
     for traj_read, traj_sub in zip(read_hdf5(test_file, 'first'), traj):
         assert (np.equal(traj_read, traj_sub).all())
     traj_no_structure = np.array([rng.randn(2, 1, 1), rng.randn(3, 1, 2)])
     pytest.raises(ValueError, write_hdf5, test_file, traj_no_structure,
                   title='second', overwrite='update')
+
+
+class XT(datetime.tzinfo):
+
+    def utcoffset(self, dt):
+        return datetime.timedelta(hours=-5)  # Eastern on standard time
+
+    def tzname(self, dt):
+        return "UTC-05:00"
+
+    def dst(self, dt):
+        return None
+
+
+def test_datetime(tmpdir):
```

Comment:
@agramfort feel free to look, if you think something is missing or there is an error let me know and I'll open another PR. One fix is on the way in https://github.com/h5io/h5io/pull/44 for Python 3.5